event_sourcery_esdb/__init__.py

```diff
@@ -27,17 +30,31 @@
 from event_sourcery_esdb.outbox import ESDBOutboxStorageStrategy
 from event_sourcery_esdb.subscription import ESDBSubscriptionStrategy
 
+Seconds: TypeAlias = PositiveFloat
+
+
+class Config(BaseModel):
+    model_config = ConfigDict(extra="forbid", frozen=True)
+
+    timeout: Seconds | None = None
+    outbox_name: str = "pyes-outbox"
+    outbox_attempts: PositiveInt = 3
+
 
 @dataclass(repr=False)
 class ESDBBackendFactory(BackendFactory):
     esdb_client: EventStoreDBClient
+    config: Config = field(default_factory=Config)
```

Comment:
Maybe protected field?

```suggestion
    _config: Config = field(default_factory=Config)
```

Reply:
I'm not sure. This is config which is immutable. I don't mind exposing it.